### ISSUE TITLE
Error during mapping DTO with private properties only with getters to a model class even by calling `ignore()`.

### ISSUE THREAD

**Issue by i-kapri** [nartc/mapper#287]
Hello again!

I am opening this issue because I have come across some error `Cannot set property friends of [object Object] which has only a getter` which is thrown after I tried to map a DTO with a private property which has got only a getter just like below:

```TypeScript
// DTO
export class UserDto {
    ...
    private _friends: number;

    get friends(): number {
        return this._friends;
    }
}
```

... and the `UserProfile` has a code line somewhere like:
```TypeScript
this.mapper.createMap(...).forMember(destination => destination.friends, ignore())
```

Do you have any idea about how to get rid of this problem?

Thank you in advance for your response!

Best regards,

Kapri

**Comment by micalevisk** [comment]
Hey @i-kapri what if you just don't use this `.forMember` method?

**Comment by nartc** [comment]
@i-kapri as @micalevisk said, does it throw the error without `.forMember()`. Since you explicitly set a Mapping Configuration for `destination.friends`, it's expected to be mapped. If it's `ignore()`, then `undefined` will be mapped to it.

**Comment by i-kapri** [comment]
Hi Mr. @nartc , hi Mr. @micalevisk  

I have managed to get rid of the error below by using some `#propertyName;` things which replace `private` but now I am stuck in the mud of this error `ERROR Error: Uncaught (in promise): TypeError: Cannot read property 'toString' of undefined
TypeError: Cannot read property 'toString' of undefined` . Does this merely have to do with the way I map the model class to the DTO back or is this a problem concerning the package itself. The afore-mentioned error is thrown when I call:

```typescript
return this.mapper.map(sourceObj, destination, source, options);
```

Thank you for your response!

Kapri

**Comment by nartc** [comment]
@i-kapri it would help if you could share your models and your mapping profile.

**Comment by i-kapri** [comment]
Hi Mr. @nartc ,

thank you for your answer.

Let's say I have this DTO object:

```typescript
export class StudentDto extends PersonDto {
    ...
    readonly girlfriendName: Readonly<string | null>;
    ...
    constructor() {
        super();
        // in the constructor I change the `writable` prop to false with C# TS-Generator so that the property is readonly.
        Object.defineProperty(this, 'girlfriendName', { { writable: false } });
    }
}
```

You have a property `isGetterOnly` of the object passed to the constructor of `AutoMap()` which I have set to true in the model class like this:

```typescript
export class Student {
    ...
    @AutoMap({ typeFn: () => String, isGetterOnly: true })
    girlfriendName: string | null | undefined; ??????? // this is readonly so I cannot assign it neither `undefined` nor sth. else.
}
```

Furthermore I cannot call `ignore()` for the property `girlfriendName` on `.forMember(...)` of `StudentProfile`. This is the problem.

I think you should check through `Object.getPropertyDescriptor(...)` whether the descriptor of the property in the destination class (DTO) is `writable: false` before you print out:

```typescript
Unmapped properties
-----------------------
....
girlfriendName
```

Thank you in advance for your answer!

Kind Regards

Kapri

**Comment by nartc** [comment]
@i-kapri Sorry for a very late response, it's been very hectic over the past 2 weeks for me. I think I have an idea of what's going on. Will get to it this weekend.

**Comment by nartc** [comment]
Fix in 4.2.1
Unmapped keys will only take writable properties.
